scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_presenter.py

```diff
@@ -557,29 +597,51 @@ def update_view_from_model(self, workspace_removed=None):
         else:
             self.selected_data = []
 
-    def check_workspaces_are_tf_asymmetry_compliant(self, workspace_list):
-        non_compliant_workspaces = [item for item in workspace_list if 'Group' not in item]
-        return False if non_compliant_workspaces else True
-
     def get_parameters_for_tf_function_calculation(self, fit_function):
         mode = 'Construct' if self.view.tf_asymmetry_mode else 'Extract'
-        workspace_list = [self.view.display_workspace] if self.view.fit_type == self.view.single_fit else self.selected_data
+        workspace_list = self.selected_data if self.view.is_simul_fit() else [
+            self.view.display_workspace]
         return {'InputFunction': fit_function,
                 'WorkspaceList': workspace_list,
                 'Mode': mode}
 
-    def create_multi_domain_function(self, function_list):
-        if not any(function_list):
-            return None
-        multi_domain_function = MultiDomainFunction()
-        for index, func in enumerate(function_list):
-            multi_domain_function.add(func)
-            multi_domain_function.setDomainIndex(index, index)
-
-        return multi_domain_function
-
     def _get_selected_groups_and_pairs(self):
         return self.context.group_pair_context.selected_groups + self.context.group_pair_context.selected_pairs
 
-    def _check_data_exists(self, guess_selection):
+    def _get_run_number_from_workspace(self, workspace_name):
+        instrument = self.context.data_context.instrument
+        run = re.findall(r'%s(\d+)' % instrument, workspace_name)
+        return run[0]
+
+    # the string following either 'Pair Asym; %s' or  Group; "
+    def _get_group_or_pair_from_workspace_name(self, workspace_name):
+        for grppair in self.context.group_pair_context.selected_groups + self.context.group_pair_context.selected_pairs:
+            grp = re.findall(r'%s' % grppair, workspace_name)
+            if len(grp) > 0:
+                return grp[0]
+
+    def _get_integer_for_grp_or_pair(self, workspace_name):
```

Comment:
I don't understand the purpose of this function. Also why does it return a double and not an int?

Reply:
The intention here is to map the group or pair names to a number, which combined with the run number can be used to sort the workspace list in a consistent manner. I agree this is poorly named, as it now returns a double in the interval (0,1). I'll rename it.